reference-implementation/lib/writable-stream.js

```diff
@@ -226,25 +240,8 @@ function WritableStreamFinishInFlightWriteWithError(stream, reason) {
 
   assert(state === 'writable');
 
-  stream._state = 'errored';
-  stream._storedError = reason;
-
-  if (stream._pendingAbortRequest === undefined) {
-    const writer = stream._writer;
-    if (writer !== undefined) {
-      if (WritableStreamCloseQueuedOrInFlight(stream) === false && stream._backpressure === true) {
-        defaultWriterReadyPromiseReject(writer, reason);
-      } else {
-        defaultWriterReadyPromiseResetToRejected(writer, reason);
-      }
-      writer._readyPromise.catch(() => {});
-    }
-  } else {
-    stream._pendingAbortRequest._reject(reason);
-    stream._pendingAbortRequest = undefined;
-  }
-
-  WritableStreamRejectPromisesInReactionToError(stream);
+  WritableStreamError(stream, error);
+  WritableStreamRejectAbortRequestIfPending(stream);
```

Comment:
Ah, it is not that easy, because WritableStreamRejectAbortRequestIfPending assumes the stream is already in the errored state. So I think it is an OK change. It would be slightly nicer for close to always be last but it still seems OK.